core/vnl/tests/test_matrix.cxx

```diff
@@ -160,18 +160,30 @@ void test_int()
        (m0.set_diagonal(vnl_vector<int>(2,2,m0values)),
         (m0.get(0,0)==7 && m0.get(1,1)==9 && m0.get(0,1)==2 && m0.get(1,0)==2)), true);
 
-  // Flip the matrix, otherwise it is the same whichever way it is flattened
-  m0.flipud();
-  // | 2 9 |
-  // | 7 2 |
-  vnl_vector<int> flat;
-  TEST("m0.flatten_row_major()",
-       (flat = m0.flatten_row_major(),
-       (flat.get(0)==2 && flat.get(1)==9 && flat.get(2)==7 && flat.get(3)==2)), true);
-  TEST("m0.flatten_column_major()",
-       (flat = m0.flatten_column_major(),
-       (flat.get(0)==2 && flat.get(1)==7 && flat.get(2)==9 && flat.get(3)==2)), true);
-  m0.flipud();
+    /////////////////////////////////////////////////////////////////
+    // Test `flatten_row_major` and `flatten_column_major` Methods //
+    /////////////////////////////////////////////////////////////////
+
+    {
+    int data[16] = { 0,  1,  2,  3,
+                     4,  5,  6,  7,
+                     8,  9, 10, 11,
+                    12, 13, 14, 15};
+
+    vnl_vector<int> flat(data, 16);
+
+    vnl_matrix<int> sq(data, 4, 4);
+    vnl_matrix<int> lg(data, 2, 8);
+    vnl_matrix<int> wd(data, 8, 2);
+
+    TEST("sq.flatten_row_major", flat.is_equal(sq.flatten_row_major(), 10e-6), true);
```

Comment:
@DVigneault   Lines 179 - 185 are causing compilation failures on the intel compiler.

error: no suitable user-defined conversion from "vnl_vector_fixed<int, 16U>" to "const vnl_vector<int>" exists


Is this intended?